File: Angular/diploma/src/models/Comunity/Reaction.ts
import { Guid } from "guid-typescript";
import { ReactionType } from "./ReactionType";
import { BaseComunityEntity } from "./BaseComunityEntity";
export class Reaction{
    public Id?:Guid;
    public EntityId!:Guid;
    public Entity?:BaseComunityEntity;
    public UserId!:number;
    public Type:ReactionType = ReactionType.None;
}

File: DiplomaProject/DataBase/Enities/Reaction.cs
namespace DiplomaProject.DataBase.Enities
{
    public class Reaction
    {
        public Guid? Id { get; set; }
        public Guid EntityId { get; set; }
        public BaseComunityEntity? Entity { get; set; }
        public ReactionType Type { get; set; }
        public long UserId { get; set; }

        public override bool Equals(object? obj)
        {
            if (obj is null)
            {
                return false;
            }
            if (obj is not Reaction)
            {
                return false;
            }
            var r = obj as Reaction;
            if (Id == r.Id && Entity == r.Entity && Type == r.Type)
                return true;
            return false;
        }
    }
}
